pywincffi/core/typesbase.py

```diff
@@ -0,0 +1,58 @@
+"""
+Types Base
+==========
+
+Provides the base types on top of which user visible types will be built.
+"""
+
+import cffi
+
+
+# pylint: disable=too-few-public-methods
+class CFFICDataWrapper(object):
+
+    """
+    Base class for exposing Python types and interfaces to pywincffi users.
+    * Wraps a CFFI cdata object in self._cdata.
+    * Delegates attribute getting/setting to self._cdata, supporting structs.
+    * Delegates item getting/setting to self._cdata, supporting arrays.
+    Attribute access is not delegated to the wrapped object if the class
+    itself contains such an attribute and that attribute is a descriptor; this
+    is in place to support @property in sub-classes.
+    """
+
+    def __init__(self, cdecl, ffi=None):
+        if ffi is None:
+            ffi = cffi.FFI()
```

Comment:
You'll want to use `dist.load()` here to retrieve the ffi object:

```python
if ffi is None:
    ffi, _ = dist.load()
```

dist.py caches an instance of `cffi.FFI` which may be loaded from the precompiled version of pywincffi.